opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,27 +12,161 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# type: ignore
+
 from sys import version_info
+from typing import Dict, Tuple, Union
+
+# This is a cache to avoid going through creating a dictionary for all entry
+# points for 3.7 every time the entry points function is called.
+result_entry_points_37 = {}
 
 # FIXME remove this when support for 3.7 is dropped.
 if version_info.minor == 7:
     # pylint: disable=import-error
-    from importlib_metadata import entry_points, version  # type: ignore
+    from importlib_metadata import EntryPoint
+    from importlib_metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+    from importlib_metadata import version
+
+    def entry_points(
+        group: str = None, name: str = None
+    ) -> Union[Tuple[EntryPoint], Dict[str, Tuple[EntryPoint]]]:
+
+        if group is None and name is None:
+
+            if not result_entry_points_37:
+
+                for entry_point in importlib_metadata_entry_points():
+
+                    if entry_point.group not in result_entry_points_37.keys():
+                        result_entry_points_37[entry_point.group] = []
+
+                    result_entry_points_37[entry_point.group].append(
+                        entry_point
+                    )
+
+                for key, value in result_entry_points_37.items():
+                    result_entry_points_37[key] = tuple(value)
+
+            return result_entry_points_37
+
+        if group is not None and name is None:
+            return tuple(
+                entry_point
+                for entry_point in importlib_metadata_entry_points(group=group)
+            )
+
+        if group is None and name is not None:
+            return tuple(
+                entry_point
+                for entry_point in importlib_metadata_entry_points(name=name)
+            )
+
+        return tuple(
+            entry_point
+            for entry_point in importlib_metadata_entry_points(
+                group=group, name=name
+            )
+        )
+
+    __all__ = [
+        "entry_points",
+        "EntryPoint",
+        "version",
+    ]
```

Comment:
`__all__` is the same values in all three branches. you can just define it once at the module level

Reply:
Right, fixed.